### DISCUSSION TITLE
Question about polls and timers

### DISCUSSION THREAD

**Discussion by sudousercom** [eternnoir/pyTelegramBotAPI#1654] [General]
Hi! I am using some scheduled messages with schedule, for remind chat member to do something at every day, but with migration on async telebot this function becomes broken, because it uses Thread for schedule cheker, iam not an programmer at all, maybe you have advise for me? the code:

```
def schedule_checker():
    while True:
        schedule.run_pending()
        sleep(1)

def reminder():
    group_ids = ["-100", "-1001"]
    for chid in group_ids:
        bot_noasync.send_message(chid, 'text')

def test_timer():
    chid = "-601"
    bot_noasync.send_message(chid, 'timer beep')

if __name__ == '__main__':
    schedule.every().day.at("16:20").do(reminder)
    schedule.every().hour.at(":00").do(test_timer)
    Thread(target=schedule_checker).start()
```

And second question is about polls, how can i automatically pin all polls in chat? In docs i found only poll handler, and nothing about how to use it. Thanks for answers!

**Comment by coder2020official** [comment]
If you are using pure async, then you will need to take a look at asyncio scheduler in apscheduler. Instead of thread tasks, you will need to use asyncio tasks. And, you cannot pin all polls in a group: you can get new polls and pin them using pin_chat_member.

**Reply by sudousercom** [reply]
I mean autopin new polls, of cource

**Reply by coder2020official** [reply]
You will need to create a handler with content_types poll, and use bot.pin_chat_message:
https://pytba.readthedocs.io/en/latest/sync_version/index.html#telebot.TeleBot.pin_chat_message

So, this will look smth like this:
```python3
from telebot import TeleBot


bot = TeleBot("TOKEN")



@bot.message_handler(content_types=['poll'])
def handle_poll(message):
    bot.pin_chat_message(
        chat_id=message.chat.id,
        message_id=message.message_id
    )


bot.infinity_polling()
```

**Reply by sudousercom** [reply]
Thank you so much, i already figured it out minutes ago, but stuck with async scheduler btw

**Reply by sudousercom** [reply]
i found example in this repo, works fine - https://github.com/eternnoir/pyTelegramBotAPI/blob/master/examples/asynchronous_telebot/timer_bot_async.py
